packages/components/src/custom-select-control/index.js

```diff
@@ -22,6 +22,12 @@ const stateReducer = (
 	{ selectedItem },
 	{ type, changes, props: { items } }
 ) => {
+	const selectedItemIndex = items.findIndex(
+		( item ) => item.key === selectedItem.key
```

Comment:
When are the items here not referentially equal? I ask because other code relies on it below.

Reply:
I haven't really looked at what Downshift is doing, but suspect they are creating a `selectedItem` object at some point, because these stopped being referentially equal when I removed `_selectedItem` from the arguments passed to `useSelect`. Where do you reckon this could cause breakage?

Reply:
https://github.com/WordPress/gutenberg/pull/22815/files#diff-406f1cf3cfd36939ce9e9607a5c408d2R152-R154

There, for example. We should stick to one way of comparing values to avoid inconsistencies.

Reply:
Got it, fixed!